### PR TITLE
integration of summation type expressions.

### LINKED ISSUE #7827: Integration of summations:
I want to compute an integral like (`n` and `N` are integers)
`integrate(summation(sin(n*x), (n,1,N)), x)`
but SymPy returns the same, i.e., it cannot interchange the summation and the integral.

For derivatives it works, though
`diff(summation(sin(n*x), (n,1,N)), x)`
returns
`Sum(n*cos(n*x), (n,1,N))` .

This feature can be useful, e.g.,  when solving differential equations with harmonic forcing.

### PR DESCRIPTION
**Pull request by Shekharrajak**
Fixes https://github.com/sympy/sympy/issues/7827

### CODE REVIEW

**Comment by Shekharrajak** [comment]
ping @jksuom @asmeurer @moorepants

**Comment by jksuom** [comment]
I am not sure it would always be the best strategy to automatically change the order of integration and summation. In any case, that should probably be done on a higher level, not as a special case in ``_eval_integral``.

**Comment by Shekharrajak** [comment]
Then Fubini's theorem says that for general fn, if` ∫∑|fn|<∞∫∑|fn|<∞ `or `∑∫|fn|<∞∑∫|fn|<∞ `(by [Tonelli theorem](https://en.wikipedia.org/wiki/Fubini%27s_theorem)  the two conditions are equivalent), then `∫∑fn=∑∫fn∫∑fn=∑∫fn`.
So I am sure that we can change the order of integration and summation.
I found this place better for alternating the function. Please suggest any other position where I can put these lines of code.

**Comment by jksuom** [comment]
There is no doubt that the order of summation and integration can be changed if the sum is finite. But I am not sure that it should be automatically done. Perhaps some kind of rewrite method for Integral would be more flexible.

File: sympy/integrals/integrals.py [hunk 1]
```diff
@@ -832,6 +834,24 @@ def _eval_integral(self, f, x, meijerg=None, risch=None, manual=None,
             if not meijerg:
                 # g(x) = Mul(trig)
                 h = trigintegrate(g, x, conds=conds)
+                if isinstance(f,Sum):
+                    f_temp =f
+                    f, limits = f.function, list(f.limits)
+                    limit = limits.pop(-1)
```
Comment (gxyd) [inline sympy/integrals/integrals.py]: Aren't you overwriting the already imported `limit` function? This is a bad practise to overwrite.

Comment (Shekharrajak) [inline sympy/integrals/integrals.py]: `limits ` is of type ` list ` with value `[(n, 1, N)]`, for further step type` tuple` is needed and that is type of `limit` with value `(n, 1, N)`.

**Comment by czgdp1807** [comment]
Closing this in favour of #18491

**Pull request closed without merging**